### PR TITLE
fix(autocomplete, filter-field): Fixes an issue with overlay positioning.

### LINKED ISSUE #451: Filter Field: Autocomplete appears above page header
# Bug Report

With an open autocomplete or range, the opened overlay appears above the sticky page-header (z-index problem). 

## Expected Behavior

I'm not entirely sure what should happen. Either the autocomplete or range box should close when scrolling or the z-index should be fixed. 

## Current Behavior

Autocomplete or range box stay open and overlay other components.

### Used Versions:

- **angular**: 8.2.12
- **@angular/cdk**: 8.2.3
- **@dynatrace/barista-components**: 4.15.1

### PR DESCRIPTION
**Pull request by tomheller**
Switched the autocomplete and filterfield overlay to use the
DtFlexibleConnectedPositionStrategy instead of the CDK
FlexibleConnectedPositionStrategy. The Dt version can deal with the
clusters page-ribbon component and in general deals with better with
overlaying elements.

Fixes #451

@thomaspink @ffriedl89 As we have to change the constructor signature here, should we consider this a breaking change?

#### Checklist

- [x] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by thomaspink** [comment]
@tomheller in the past we handled it as a breaking change (unless the additional params are optional)

**Comment by tomheller** [comment]
@thomaspink What do you think, should we hold off on releasing this until we deliver v6.0.0 with Angular 9 support or schedule this for a minor version. The added constructor parameters are not optional, but are only in internal components like the autocomplete and the range component (part of the filter field). 

I don't have a strong opinion on how we should proceed.

**Comment by thomaspink** [comment]
As it is only an internal component, I am fine with releasing it as non-breaking. I thing the benefit of releasing it earlier (via a patch release) outweighs the breaking part of it (as it is not used directly by the consumer).

**Comment by tomheller** [comment]
@thomaspink @lukasholzer @ffriedl89 I have made the newly added Injections optional and have built a switch check into the creation of the new PositionStrategy. This should no longer be considered a breaking change. 
I have added breaking change comments for version 7 to remove the fallback to the cdk FlexibleConnectedPositionStrategy.

File: components/filter-field/src/filter-field-range/filter-field-range-trigger.ts [hunk 1]
```diff
@@ -139,6 +153,11 @@ export class DtFilterFieldRangeTrigger implements OnDestroy {
     private _changeDetectorRef: ChangeDetectorRef,
     zone: NgZone,
     renderer: Renderer2,
+    @Optional() private _viewportRuler: ViewportRuler,
```
Comment (thomaspink) [inline components/filter-field/src/filter-field-range/filter-field-range-trigger.ts]: to make it non breaking pls make the prop itself also optional, not only via DI.
```suggestion
    @Optional() private _viewportRuler?: ViewportRuler,
```

Also annotate it as a breaking fo 7.0.0 to remove it again

File: components/filter-field/src/filter-field-range/filter-field-range-trigger.ts [hunk 2]
```diff
@@ -139,6 +153,11 @@ export class DtFilterFieldRangeTrigger implements OnDestroy {
...
+    @Optional() private _platform: Platform,
```
Comment (thomaspink) [inline components/filter-field/src/filter-field-range/filter-field-range-trigger.ts]: Same here
```suggestion
    @Optional() private _platform?: Platform,
```

File: components/filter-field/src/filter-field-range/filter-field-range-trigger.ts [hunk 3]
```diff
@@ -139,6 +153,11 @@ export class DtFilterFieldRangeTrigger implements OnDestroy {
...
+    @Optional() private _overlayContainer: OverlayContainer,
```
Comment (thomaspink) [inline components/filter-field/src/filter-field-range/filter-field-range-trigger.ts]:
```suggestion
    @Optional() private _overlayContainer?: OverlayContainer,
```

File: components/filter-field/src/filter-field-range/filter-field-range-trigger.ts [hunk 4]
```diff
@@ -139,6 +153,11 @@ export class DtFilterFieldRangeTrigger implements OnDestroy {
...
+    // tslint:disable-next-line:no-any
+    @Optional() @Inject(DOCUMENT) private _document: any,
```
Comment (thomaspink) [inline components/filter-field/src/filter-field-range/filter-field-range-trigger.ts]:
```suggestion
    @Optional() @Inject(DOCUMENT) private _document?: any,
```

**Pull request merged**